libs/api/domains/driving-license/src/lib/client/drivingLicense.api.ts

```diff
@@ -23,13 +32,29 @@ export class DrivingLicenseApi {
     this.secret = secret
   }
 
-  async requestApi(url: string) {
+  headers() {
+    return {
+      'X-Road-Client': this.xroadClientId,
+      SECRET: this.secret,
+      Accept: 'application/json',
+    }
+  }
+
+  async postApi(url: string, body: {}) {
     const res = await fetch(`${this.xroadApiUrl}/${url}`, {
       headers: {
-        'X-Road-Client': this.xroadClientId,
-        SECRET: this.secret,
-        Accept: 'application/json',
+        ...this.headers(),
+        'Content-Type': 'application/json',
       },
+      method: 'POST',
+      body: JSON.stringify(body),
+    })
+    return res.json()
+  }
```

Comment:
Are there OpenAPI/Swagger docs for this api? Would be nice to use a generated client for consistency.

Reply:
there is - i haven't looked into how to generate the client from those.. i had an attempt at it but since parts of it were also already implemented, and i was stilling getting the hang of things i abandoned it.

but i'd be happy to do so as a followup as well